app/src/main/java/com/mk/tasky/agenda/data/AgendaRepositoryImpl.kt

```diff
@@ -222,32 +218,26 @@ class AgendaRepositoryImpl(
         // Edge Cases:
         // - If we get event remotely with different attendees, we should remove all from db and add the new ones, since we could have outdated users
         insertEventLocally(event)
-        saveEventRemotely(event, isEdit).onFailure {
-            // TODO: Save id on db to later sync with server
-            println(it.message)
-        }
-    }
 
-    private suspend fun saveEventRemotely(event: AgendaItem.Event, isEdit: Boolean) = resultOf {
         val moshi = Moshi.Builder().build()
         val jsonAdapter = moshi.adapter(EventDto::class.java)
         val json: String = jsonAdapter.toJson(event.toDto())
 
-        val photos = event.photos.mapIndexed { index, photo ->
-            val uri = Uri.parse(photo.location)
-            val extension = photoExtensionParser.extensionFromUri(uri)
-            val uriBytes = photoByteConverter.uriToBytes(uri)
-
-            MultipartBody.Part.createFormData(
-                "photo$index",
-                filename = UUID.randomUUID().toString() + "." + extension,
-                uriBytes.toRequestBody()
-            )
-        }
-        // TODO: Add isEdit
-        api.createEvent(
-            body = MultipartBody.Part.createFormData("create_event_request", json),
-            files = photos
-        )
+        val photoLocations = event.photos.map { it.location }
+        val uploaderWorker = OneTimeWorkRequestBuilder<EventUploaderWorker>().setConstraints(
+            Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+
+        ).setInputData(
+            Data.Builder()
+                .putString(EventUploaderWorkerParameters.EVENT_JSON, json)
+                .putBoolean(EventUploaderWorkerParameters.IS_EDIT, isEdit)
+                .putStringArray(
+                    EventUploaderWorkerParameters.EVENT_PHOTO_ARRAY,
+                    photoLocations.toTypedArray()
+                ).build()
+        ).build()
```

Comment:
It can also make sense to assign a backoff time. Imagine the server is not reachable or returns an error for a short time. Then you'd like to retry this request at a later time.
For example, with `BackoffPolicy.EXPONENTIAL` the time till the next request is increased exponentially, so you quickly check if it was just a one-time server issue (since it'll start to retry frequently), but if it keeps on failing, the time till the next request will grow exponentially which makes sense in this case.

Reply:
Makes sense. Will add it!